game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/EndPoint.java

```diff
@@ -119,16 +116,9 @@ boolean removeImplementor(final Object implementor) {
 
   private List<RemoteMethodCallResults> invokeMultiple(
       final RemoteMethodCall call, final INode messageOriginator) {
-    // copy the implementors
-    final List<Object> implementorsCopy;
-    synchronized (implementorsMutex) {
-      implementorsCopy = new ArrayList<>(implementors);
-    }
-    final List<RemoteMethodCallResults> results = new ArrayList<>(implementorsCopy.size());
-    for (final Object implementor : implementorsCopy) {
-      results.add(invokeSingle(call, implementor, messageOriginator));
-    }
-    return results;
+    return implementors.stream()
```

Comment:
Important, is there *any* chance we will invoke this method while implementors are being removed? If so we could hit a concurrent modification exception. Have you considered creating an implementor collection copy? Since we are copying references of a couple dozen objects, it should be a negligibly cheap operation.

Reply:
Unlikely we'll invoke both at the same time, but implementors is a thread safe implementation, which effectively creates a defensive copy whenever something is written to it so no worries there.

Reply:
Cool, thanks :+1: